lib/IRGen/GenType.cpp

```diff
@@ -712,16 +726,18 @@ void FixedTypeInfo::storeEnumTagSinglePayload(IRGenFunction &IGF,
   Builder.CreateStore(payloadIndex, payloadIndexAddr);
   Address extraTagIndexAddr = IGF.createAlloca(int32Ty, Alignment(4));
   Builder.CreateStore(extraTagIndex, extraTagIndexAddr);
-  // TODO: big endian
+
+  Address alignedPayloadIndexAddr = alignAddress(IGF, payloadIndexAddr, llvm::ConstantInt::getSigned(IGM.Int32Ty, std::min(Size(4U).getValue(), fixedSize.getValue())));
   Builder.CreateMemCpy(
       valueAddr,
-      Builder.CreateBitCast(payloadIndexAddr, IGM.Int8PtrTy),
+      alignedPayloadIndexAddr,
       std::min(Size(4U), fixedSize));
```

Comment:
You should definitely factor out this sub-expression.

Reply:
At the very least, the `std::min` part of that expression, but you're also using the constant in a few places so that could come out too.  I'm just wary of a magic number being copied and pasted three times.